File: source/frontend/projects/pulse/src/app/api/models/student-session-details-dto.ts
/* tslint:disable */
/* eslint-disable */
import { InstructorEmoticonListItemDto } from './instructor-emoticon-list-item-dto';
import { SessionCheckinDetailsDto } from './session-checkin-details-dto';
import { SessionQuestionDetailsDto } from './session-question-details-dto';
export interface StudentSessionDetailsDto {
  activeCheckin?: null | SessionCheckinDetailsDto;
  code: string;
  emoticons: Array<InstructorEmoticonListItemDto>;
  id: string;
  questions?: null | Array<SessionQuestionDetailsDto>;
}

File: source/backend/projects/Pulse.Api/Models/Students/StudentSessionDetailsDto.cs
namespace Pulse.Api.Models.Students
{
    using System;
    using System.ComponentModel.DataAnnotations;
    using Emoticons;
    using Sessions;

    public class StudentSessionDetailsDto
    {
        [Required]
        public Guid Id { get; set; }

        [Required]
        public string Code { get; set; }

        [Required]
        public InstructorEmoticonListItemDto[] Emoticons { get; set; }

        public SessionCheckinDetailsDto ActiveCheckin { get; set; }

        public SessionQuestionDetailsDto[] Questions { get; set; }
    }
}
